src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,120 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            newNode.next = head;
+            if (head != null) head.prev = newNode;
+            head = newNode;
+            if (size == 0) tail = newNode;
+        } else if (index == size) {
+            newNode.prev = tail;
+            if (tail != null) tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> nextNode = findNode(index);
+            Node<T> prevNode = nextNode.prev;
+            newNode.next = nextNode;
+            newNode.prev = prevNode;
+            if (prevNode != null) prevNode.next = newNode;
+            nextNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        list.forEach(this::add);
```

Comment:
The `addAll` method should check if the provided list is null to avoid a NullPointerException.